core/src/main/java/dk/alexandra/fresco/lib/compare/bool/eq/AltBinaryEqualityProtocol.java

```diff
@@ -80,16 +70,17 @@ public ProtocolProducer nextProtocolProducer() {
 			xnorOuts[0] = out;
 			blb.addProtocolProducer(new TreeProtocol(this));
 			round++;
-		} else if (round == 3){
+		} else {*/
 			return null;
```

Comment:
null is dangerous - it will not function, maybe remove?

Reply:
Deleting this too since we never refer to it, and if we ever want to optimize, we should do this on the primary implementation. No reason to have a slow one.

Reply:
Oh, but if you have a good idea of how to do a tree based approach using the new arch, I would love to hear about this. (Maybe I'll just pop by your office tomorrow)

Reply:
Oh we do and are looking forward to your vist.

However, SumSIntList and ProductSIntList are both doing a tree based collation for 2-input gates

Reply:
I actually already made the tree based protocol, it should be commit in a second.